classes/class-woothemes-sensei-certificates.php

```diff
@@ -1483,24 +1483,36 @@ public function enqueue_block_editor_assets() {
 	 * @return string Block HTML.
 	 */
 	public function update_view_certificate_button_url( $block_content, $block ): string {
+		$class_name = 'view-certificate';
+
 		// phpcs:ignore WordPress.Security.NonceVerification.Recommended -- Only used if the learner completed the course.
 		$course_id = isset( $_GET['course_id'] ) ? (int) $_GET['course_id'] : false;
 
-		// Check that the course ID exists and that the user has completed the course.
 		if (
+			// Check that the course ID exists and that the user has completed the course.
 			! $course_id
 			|| ! get_current_user_id()
 			|| 'course' !== get_post_type( $course_id )
 			|| ! Sensei_Utils::user_completed_course( $course_id, get_current_user_id() )
+
+			// Check that the block is a core/button and it contains the respective class name.
+			|| ! isset( $block['blockName'] )
```

Comment:
This check is also in `update_button_block_url`, but I needed to test it earlier to return the block when it's not the respective block, and empty when it doesn't have the template.

Reply:
Would it be worth extracting these checks to a method that could be used in both places?

Reply:
Since it's a method from the core, we'd need to write this helper method there, and also replicate it here in case their core plugin is not with the version with the helper function. So I don't think it's worth doing now. WDYT?

Reply:
Right, gotcha. Ok that's fine by me 👍